example/flutter_example/lib/highlighted_text.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+
+/// The core function for handling text highlighting
+TextSpan highlightedTextToSpan(
+  String text, {
+  required String preTag,
+  required String postTag,
+  TextStyle? highlightedStyle,
+}) {
+  final preIndex = text.indexOf(preTag);
+  final postIndex = text.indexOf(postTag);
+  if (preIndex < 0 || postIndex < 0) {
+    //if the text doesn't contain pre/post tags, there is no need for highlighting
+    return TextSpan(text: text);
+  } else {
+    //before the pre tag should be normal text
+    final beforePre = preIndex == 0 ? null : text.substring(0, preIndex);
+    //after the post tag might be normal or highlighted
+    final afterPost = text.substring(postIndex + postTag.length);
+    //between the Pre and Post tags is the highlighted text
+    final between = text.substring(preIndex + preTag.length, postIndex);
+    return TextSpan(children: [
+      if (beforePre != null && beforePre.isNotEmpty) TextSpan(text: beforePre),
+      TextSpan(text: between, style: highlightedStyle),

```

Comment:
I was thinking about this, and it is very hard to use a feature that is already "implemented" on the meilisearch side.

Do you have any idea to make it easier for the final user?

Reply:
depends on where you want to make it easier, is it in the meilisearch backend itself, or in the sdk ?

Reply:
In the SDK, because the usage in the backend is fairly easy, you had to go over the string, looking for specific pieces and adding the span just to make it work.

Reply:
I have updated the example to include this
`parseHighligtableString` is a platform agnostic function that transforms a highlightable string to parts, where each part has an `isHighlighted` property